github_workflow.md

```diff
@@ -0,0 +1,153 @@
+# GitHub Milestone and Issue Workflow
+
+This purpose of this document is to describe our development workflow on
+GitHub; in particular, how we use milestones and issues. We recognize that our
+many subprojects may use variations of these ideas. This particular workflow is
+based on the practices from the JupyterLab repo
+(https://github.com/jupyterlab/jupyterlab) and are optional for Jupyter
+subprojects to follow.
+
+## Milestones
+
+We use Github milestones to track issues and pull requests related to particular
+releases of a subproject (https://github.com/jupyterlab/jupyterlab/milestones).
+
+* These milestones should be labeled with the version numbers of the releases
+  (`4.3`, `5.0`, etc.).
+* Subprojects should only have a few numbered milestones. Beyond one or two
+  future releases, we have found that detailed milestones become meaningless.
+* Subprojects should also have a `Backlog` milestone for issues and pull
+  requests that have not been assigned to a particular release milestone. This
+  also includes new feature requests ("wishlist") issues that we haven't
+  committed to implementing. Essentially, `Backlog` means "at some point in the
+  future, maybe."
+* Subprojects should have a `No Action` milestone to indicate issues or pull
+  requests that were closed with no action or work being performed. In general
+  we try to close issues/PRs with this milestone.
+* The only time an issue should have no milestone is early on in the triage
+  process when it still has the `status:Needs Info` label.
+
+## Labels
+
+We use GitHub labels to organize issues and pull requests
+(https://github.com/jupyterlab/jupyterlab/labels). The Github issues namespace
+is completely unstructured; each label is simply a name with an associated
+color. This unstructured nature, makes it difficult to organize work
+effectively. To address, this challenge, we are started to use a more formal
+label schema/taxonomy.
+
+First, every label should have one of four prefixes:
+
+1. `type:`
+2. `component:`
+3. `cat:`
+4. `status:`
+
+All issues of a given prefix should have the same color to make visual
+identification easy.
+
+### `type:`
+
+The `type:` prefix indicates the type of issue or pull request. There are four
+types:
+
+* `type:bug`
+* `type:enhancement`
+* `type:maintenance`
```

Comment:
Yeah, we definitely run into cases where something is sort of both/neither bug/new feature. However, being able to do structured queries in a manner that has only one `type:` prefixed label has been super helpful for us - as it let's us get an overview of how a milestone is doing relative to bugs/new features. Because of this, we are semi-forcing ourselves to pick between bug or enchancement (or question or mainenance).

The maintenance type is used for refactoring and other maintenance that doesn't fix a bug and isn't a new feature. Sort of "cleanup"...